asynction/security.py

```diff
@@ -360,14 +360,25 @@ def security_handler(request: Request) -> SecurityInfo:
     return security_handler
 
 
+def resolve_security(
+    channel_security: Sequence[SecurityRequirement],
+    server_security: Sequence[SecurityRequirement],
+) -> Sequence[InternalSecurityRequirement]:
+    unpacked_channel_security = unpack_security_requirements(channel_security)
+    unpacked_server_security = unpack_security_requirements(server_security)
+
+    return [*unpacked_channel_security, *unpacked_server_security]
```

Comment:
```suggestion
    return unpacked_channel_security or unpacked_server_security
```
I think that the channel requirements should overwrite (and not override) the server requirements. Not sure what would be most useful in the long term, but It is a safer design decision to go with overwrite, since we can switch it to override in a backwards compatible manner (while the inverse cannot happen).

And this or statement can happen on the register method level rather than here (see comment below).